cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -80,15 +79,36 @@ func DeployedDNSAddon(client clientset.Interface) (kubeadmapi.DNSAddOnType, stri
 	}
 }
 
+// deployedDNSReplicas returns the replica count for the current DNS deployment
+func deployedDNSReplicas(client clientset.Interface, replicas int32) (*int32, error) {
```

Comment:
This func is a bit confusing at first sight. It is supposed to return the number of DNS replicas, but if it fails it returns a default value (supplied as a param). It seems like a copy of `DeployedDNSAddon` too.
Can we modify `DeployedDNSAddon` to return the replica count too and reuse that instead?

Reply:
It is a copy. I was hesitant to modify `DeployedDNSAddon` as it's exported and referenced in a few places and I didn't want to have ignored return values in other places. But, I also know we don't want to have duplicate code. Happy to hear your thoughts on the best approach for this @rosti